src/controllers/plugins/api/JSONParse.cpp

```diff
@@ -0,0 +1,379 @@
+            .index = 0,
+            .isObject = 0,
+        });
+        if (elements.size() > MAX_NESTING)
+        {
+            ec = makeCode(ParseError::TooMuchNesting);
+            return false;
+        }
+
+        lua_checkstack(this->state, 2);  // grow stack by 2 (table, value)
+        lua_newtable(this->state);
+
+        return true;
+    }
+
+    bool on_array_end(size_t /* nElements */, boost::system::error_code &ec)
+    {
+        if (this->elements.empty())
+        {
+            ec = makeCode(ParseError::NoElements);
+            assert(false && "We shouldn't hit this");
+            return false;
+        }
+        assert(!this->elements.back().isObject);
+        this->elements.pop_back();
+
+        return this->appendValue(ec);
+    }
+
+    bool on_key_part(boost::json::string_view sv, std::size_t /* n */,
+                     boost::system::error_code & /* ec */)
+    {
+        this->buffer.append(sv);
+        return true;  // no push - we'll get a on_key call
+    }
+
+    bool on_key(boost::json::string_view sv, size_t /* n */,
+                boost::system::error_code & /* ec */)
+    {
+        if (this->buffer.empty())
+        {
+            lua_pushlstring(this->state, sv.data(), sv.size());
+        }
+        else
+        {
+            this->buffer.append(sv);
+            lua_pushlstring(this->state, this->buffer.c_str(),
+                            this->buffer.size());
+            this->buffer.clear();
+        }
+
+        // no appendValue (wait for the value)
+        return true;
+    }
+
+    // scalar values
+    bool on_string_part(boost::json::string_view sv, size_t /* n */,
+                        boost::system::error_code & /* ec */)
+    {
+        this->buffer.append(sv);
+        return true;  // no appendValue - we'll get a on_string call
+    }
+
+    bool on_string(boost::json::string_view sv, size_t /* n */,
+                   boost::system::error_code &ec)
+    {
+        if (this->buffer.empty())
+        {
+            lua_pushlstring(this->state, sv.data(), sv.size());
+        }
+        else
+        {
+            this->buffer.append(sv);
+            lua_pushlstring(this->state, this->buffer.c_str(),
+                            this->buffer.size());
+            this->buffer.clear();
+        }
+
+        return this->appendValue(ec);
+    }
+
+    bool on_number_part(boost::json::string_view /* part */,
+                        boost::system::error_code & /* ec */)
+    {
+        // unhandled
+        return true;
+    }
+
+    bool on_int64(int64_t i, boost::json::string_view /* source */,
+                  boost::system::error_code &ec)
+    {
+        lua_pushinteger(this->state, static_cast<lua_Integer>(i));
+        return this->appendValue(ec);
+    }
+
+    bool on_uint64(uint64_t i, boost::json::string_view /* source */,
+                   boost::system::error_code &ec)
+    {
+        lua_pushinteger(this->state, static_cast<lua_Integer>(i));
```

Comment:
`lua_Integer` is signed which means that parsing `18446744073709551615` ($2^{64}-1$) yields `-1`. I think this should be converted to a floating point like JS does to garble the data less (consider `json.stringify(json.parse(18446744073709551615)` case). I'd like to have a test added for this.